### ISSUE TITLE
Added bcrypt as password hashing method instead of md5

### ISSUE THREAD

**Issue by mickaelandrieu** [backbee/BackBee#290]
This need a new dependency on (https://github.com/ircmaxell/password_compat) because bcrypt function is only available for PHP > 5.5

md5 is a very weak encoding method, this ia matter of minutes to broke this kind of databases. Now BackBee is open source, hackers can easily break our databases because they already know how the system works.

**Comment by mickaelandrieu** [comment]
For 1.0 release, in my opionon we should use a factory to choose the correct encoder like in Symfony original "stuff". "Don't re-invent the wheel"

**Comment by ndufreche** [comment]
The encryption normally have to be customizable in configuration is for that we use the EncoderFactory.

**Comment by mickaelandrieu** [comment]
@ndufreche you are right (https://github.com/backbee/BackBee/blob/master/Rest/Controller/UserController.php#L211) but in installer we only md5. So how BackBee can choose another encoder if the first user is "hard coded" ?

**Comment by ndufreche** [comment]
@mickaelandrieu ok, we have to change that remove the md5 default encoding from security.yml

**Comment by mickaelandrieu** [comment]
@ndufreche maybe can we suggest an encoding method in the installer ?
 - md5 
 - sha1
 - bcrypt (recommended)

This way we write this choose in the ``security.yml`` file of website application.

**Comment by ndufreche** [comment]
If bcrypt is recommended, we can set it by default. I'm not on the fact of asking what type of encryption the user want during the installation.

**Comment by mickaelandrieu** [comment]
Agree. If we need bcrypt we have a new dependency on ircmaxwell/password_compat (Symfony and Laravel both have too)

**Comment by ndufreche** [comment]
We no need to add it, is already in the security-core dependencies.

**Comment by mickaelandrieu** [comment]
ok ! (good news :) )
